syntax = "proto3";

package transport_catalogue_serialize;

import "map_renderer.proto";

message Coordinates
{
    double lat = 1;
    double lng = 2;
}

message Stop
{
    bytes name = 1;
    Coordinates coordinates = 2;
}

message Bus
{
    bytes name = 1;
    repeated Stop stops = 2;
    bool roundtrip = 3;
}

message Distance
{
    bytes from = 1;
    bytes to = 2;
    int64 distance = 3;
}

message TransportCatalogue
{
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
	RenderSettings render_settings = 4;
}